### PR TITLE
🌱 E2e: Support running Ironic with MariaDB

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:

- Support running Ironic with MariaDB
- Use new deploy script args instead of positional booleans
- Add keepalived variable to e2e config for consistency

Related:
- https://github.com/metal3-io/baremetal-operator/pull/1196
- https://github.com/metal3-io/metal3-dev-env/pull/1127

### CODE REVIEW

**Comment by lentzi90** [comment]
Successful tests with mariadb :tada: 

- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_test_centos_parallel/74/
- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_test_ubuntu_parallel/135/

**Comment by lentzi90** [comment]
Running tests without mariadb enabled now.
/test-ubuntu-e2e-parallel-main
/test-centos-e2e-parallel-main

**Comment by lentzi90** [comment]
Successful tests without mariadb:
- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_test_centos_parallel/75/
- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_test_ubuntu_parallel/136/

**Review by kashifest — APPROVED** [review]
/hold

**Comment by lentzi90** [comment]
Dropped the test commit now that the metal3-dev-env PR has been merged. Will rerun tests. Should be possible to get green tests on this already before BMO is merged. This way we can test everything properly before merging the BMO PR.

File: test/e2e/config/e2e_conf.yaml
Comment (lentzi90) [reply test/e2e/config/e2e_conf.yaml:128]: No. I think it makes more sense like this. Upgrading the database is not really the concern of the Metal3 stack. Besides, all our tests use sqlite. If we add a test that use mariadb at some point we can consider adding this back again, but for now I think it is unnecessary complicated and confusing to keep it.

**Comment by adilGhaffarDev** [comment]
@lentzi90 should I do manual cherry pick ?

**Comment by lentzi90** [comment]
Maybe :slightly_frowning_face: I'm just fearing it could be pretty hard. Worth a try though

**Comment by mboukhalfa** [comment]
I should not backport either to 1.3 because it is running with old bmo version v0.2.0 https://github.com/metal3-io/metal3-dev-env/blob/9244e805820b996fa166dc38977f8126d2ad3520/vars.md?plain=1#L79

**Pull request merged**
